cache/src/main/java/io/envoyproxy/controlplane/cache/ConfigWatcher.java

```diff
@@ -20,10 +20,12 @@
    * @param request the discovery request (node, names, etc.) to use to generate the watch
    * @param knownResourceNames resources that are already known to the caller
    * @param responseConsumer the response handler, used to process outgoing response messages
+   * @param hasClusterChanged indicates if EDS should be send. Works in ads mode only.
    */
   Watch createWatch(
       boolean ads,
       DiscoveryRequest request,
       Set<String> knownResourceNames,
-      Consumer<Response> responseConsumer);
+      Consumer<Response> responseConsumer,
+      boolean hasClusterChanged);
```

Comment:
That's a breaking change in the API, is there a way do it backward compatible?

Reply:
I don't think it should be backward compatible. I think if someone use this interface in a custom Cache implementation, it is better that the build will fail after upgrade. It will force to take the new parameter into consideration and handle the potential bug which we handled in our implementation.

If custom implementation really doesn't care about this, they can simply ignore the new parameter. But it will be conscious decision, not automatic one.